src/components/sidenav/sidenav.ts

```diff
@@ -0,0 +1,353 @@
+import {
+  AfterContentInit,
+  Component,
+  ContentChildren,
+  ElementRef,
+  EventEmitter,
+  Host,
+  HostBinding,
+  HostListener,
+  Input,
+  View,
+  ViewEncapsulation,
+  OnChanges,
+  Optional,
+  Output,
+  Query,
+  QueryList,
+  SimpleChange,
+  Type
+} from 'angular2/core';
+import {BaseException} from 'angular2/src/facade/exceptions';
+import {CONST_EXPR} from 'angular2/src/facade/lang';
+import {Dir} from '../../directives/dir/dir';
+import {OneOf} from '../../core/annotations/one-of';
+
+
+/**
+ * Exception thrown when a MdSidenavLayout is missing both sidenavs.
+ */
+export class MdMissingSidenavException extends BaseException {}
+
+/**
+ * Exception thrown when two MdSidenav are matching the same side.
+ */
+export class MdDuplicatedSidenavException extends BaseException {
+  constructor(align: string) {
+    super(`A sidenav was already declared for 'align="${align}"'`);
+  }
+}
+
+
+/**
+ * <md-sidenav> component.
+ *
+ * This component corresponds to the drawer of the sidenav.
+ *
+ * Please refer to README.md for examples on how to use it.
+ */
+@Component({
+  selector: 'md-sidenav',
+  template: '<ng-content></ng-content>',
+})
+export class MdSidenav {
+  /** Alignment of the sidenav (direction neutral); whether 'start' or 'end'. */
+  @Input() @OneOf(['start', 'end']) align: string = 'start';
+
+  /** Mode of the sidenav; whether 'over' or 'side'. */
+  @Input() @OneOf(['over', 'push', 'side']) mode: string = 'over';
+
+  /** Whether the sidenav is opened. */
+  @Input('opened') private opened_: boolean;
+
+  /** Event emitted when the sidenav is being opened. Use this to synchronize animations. */
+  @Output('open-start') onOpenStart = new EventEmitter<void>();
+
+  /** Event emitted when the sidenav is fully opened. */
+  @Output('open') onOpen = new EventEmitter<void>();
+
+  /** Event emitted when the sidenav is being closed. Use this to synchronize animations. */
+  @Output('close-start') onCloseStart = new EventEmitter<void>();
+
+  /** Event emitted when the sidenav is fully closed. */
+  @Output('close') onClose = new EventEmitter<void>();
+
+
+  /**
+   * Constructor.
```

Comment:
The word "Constructor" in the comment is redundant when the _syntax_ is `constructor`